### PR TITLE
Feature( user)add third party user、login for phone number、remove user roles

### PR DESCRIPTION
**Pull request by 15168440402**

### CODE REVIEW

File: src/BuildingBlocks/StackSdks/Auth/Masa.BuildingBlocks.StackSdks.Auth/Service/IUserService.cs [hunk 1]
```diff
@@ -17,11 +17,13 @@ public interface IUserService
...
     Task<long> GetTotalByTeamAsync(Guid teamId);
 
-    Task<UserModel?> AddAsync(AddUserModel user);
+    Task<UserModel> AddAsync(AddUserModel user);
 
-    Task<UserModel?> UpsertThirdPartyUserAsync(UpsertThirdPartyUserModel user);
+    Task<UserModel> UpsertThirdPartyUserAsync(UpsertThirdPartyUserModel user);
 
-    Task<UserModel?> UpsertAsync(UpsertUserModel user);
+    Task<UserModel> AddThirdPartyUserAsync(AddThirdPartyUserModel user,bool whenExistReturn);
```
Comment (zhenlei520) [inline src/BuildingBlocks/StackSdks/Auth/Masa.BuildingBlocks.StackSdks.Auth/Service/IUserService.cs:24]: format code

File: src/BuildingBlocks/StackSdks/Auth/Masa.BuildingBlocks.StackSdks.Auth/Service/IUserService.cs [hunk 2]
```diff
@@ -64,5 +66,9 @@ public interface IUserService
     Task<bool> DisableUserAsync(DisableUserModel user);
 
     Task<List<UserSimpleModel>> GetListByAccountAsync(IEnumerable<string> accounts);
+
+    Task<bool> LoginForPhoneNumberAsync(LoginForPhoneNumber login);
```
Comment (doddgu) [inline src/BuildingBlocks/StackSdks/Auth/Masa.BuildingBlocks.StackSdks.Auth/Service/IUserService.cs:70]: LoginForPhoneNumberAsync? LoginByPhoneNumberAsync?

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs [hunk 3]
```diff
@@ -234,5 +234,23 @@ public async Task<bool> UpdatePhoneNumberAsync(UpdateUserPhoneNumberModel user)
...
+        return await _caller.PostAsync<bool>(requestUri, login);
+    }
+
+    public async Task RemoveUserRolesAsync(RemoveUserRolesModel user)
+    {
+        var requestUri = $"api/user/removeUserRoles";
+        await _caller.DeleteAsync(requestUri, user);
+    }
+
+    public async Task<UserModel> AddThirdPartyUserAsync(AddThirdPartyUserModel user, bool whenExistReturn = true)
```
Comment (doddgu) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs:250]: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=231&id=masastack_MASA.Framework&open=AYMVnTG2fHq-SUAraI4y

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/StackSdks/Tests/Masa.Contrib.StackSdks.Auth.Tests/_Imports.cs [hunk 4]
```diff
@@ -11,3 +11,4 @@
 global using Microsoft.VisualStudio.TestTools.UnitTesting;
 global using Moq;
 global using System.Text.Json;
+global using Masa.Contrib.StackSdks.Auth.Service;
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Tests/Masa.Contrib.StackSdks.Auth.Tests/_Imports.cs:14]: global using sort

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs (hunk 3 above)
Comment (15168440402) [reply src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs:250]: no need to change

Comment (doddgu) [reply src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs:250]: @zhenlei520

Comment (zhenlei520) [reply src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs:250]: public async Task<UserModel> AddThirdPartyUserAsync(AddThirdPartyUserModel user, bool whenExistReturn = true)

-> public async Task<UserModel> AddThirdPartyUserAsync(AddThirdPartyUserModel user, bool whenExistReturn)

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs (hunk 3 above)
Comment (15168440402) [reply src/Contrib/StackSdks/Masa.Contrib.StackSdks.Auth/Service/UserService.cs:250]: changed

**Review by zhenlei520 — APPROVED** [review]
@doddgu LGTM

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**